app.js

```diff
@@ -0,0 +1,130 @@
+//For secure connection:
+// const fs = require('fs');
+const { Pool } = require("pg");
+
+// Configure the database connection.
+
+const config = {
+  user: "max",
+  password: "roach",
+  host: "localhost",
+  database: "bank",
+  port: 26257,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+  //For secure connection:
+  /*ssl: {
+        ca: fs.readFileSync('/certs/ca.crt')
+            .toString()
+    }*/
+};
+
+// Create a connection pool
+
+const pool = new Pool(config);
+
+// Wrapper for a transaction.  This automatically re-calls the operation with
+// the client as an argument as long as the database server asks for
+// the transaction to be retried.
+
+async function retryTxn(n, max, client, operation, callback) {
+  await client.query("BEGIN;");
+  while (true) {
+    n++;
+    if (n === max) {
+      throw new Error("Max retry count reached.");
+    }
+    try {
+      await operation(client, callback);
+      await client.query("COMMIT;");
+      return;
+    } catch (err) {
+      if (err.code != "40001") {
+        return callback(err);
+      } else {
+        console.log("Transaction failed. Retrying transaction.");
+        console.log(err.message);
```

Comment:
You could use `console.error()` if you want to make it show up angry :)

Reply:
I think we want to make it show up not-so angry, since it's going to keep retrying the transaction. @rafiss Does that sound right?